### PR TITLE
Cm4703 options to request twitter and mailchimp still available

### PR DESCRIPTION
**Pull request by juanignaciosl**
This closes CartoDB/cartodb-management/issues/4703.

@xavijam CR frontend, please.
@gfiorav CR backend, please.

### CODE REVIEW

**Comment by juanignaciosl** [comment]
@xavijam, @gfiorav next CR, please.

File: NEWS.md [hunk 1]
```diff
@@ -59,6 +59,7 @@ sudo make install
 * Configurable path for public uploads:
   * `RAILS_PUBLIC_UPLOADS_PATH`. Example: /var/carto/assets. Defaults to `env_app_config[:importer]["uploads_path"]`
   This will store user uploaded assets at `#{RAILS_PUBLIC_UPLOADS_PATH}/uploads` (needed for backwards compatibility).
+* Don''t display Twitter or MailChimp if user can''t import it.
```
Comment (xavijam) [inline NEWS.md]: Minor: double '.

File: lib/assets/javascripts/cartodb/common/dialogs/create/listing/import_options.js [hunk 2]
```diff
@@ -34,7 +34,7 @@ module.exports = {
   },
   GDrive:   {
     className: ImportService,
-    enabled: function (config) { return !!config.get('oauth_gdrive'); },
+    enabled: function (config, user_data) { return !!config.get('oauth_gdrive'); },
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb/common/dialogs/create/listing/import_options.js]: If you don't mind  @juanignaciosl, could you we userData with camelcase?

File: lib/assets/javascripts/cartodb/common/dialogs/create/listing/imports_view.js [hunk 3]
```diff
@@ -80,7 +80,7 @@ module.exports = cdb.core.View.extend({
     var tabTemplate = cdb.templates.getTemplate('common/views/create/listing/import_tab');
 
     _.each(this._IMPORT_OPTIONS, function(t) {
-      if (!_.isEmpty(t) && t.enabled(cdb.config)) {
+      if (!_.isEmpty(t) && t.enabled(cdb.config, this.user_data)) {
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb/common/dialogs/create/listing/imports_view.js]: Same as mentioned above.

File: lib/assets/javascripts/cartodb3/components/modals/add-layer/content/imports/import-options.js [hunk 4]
```diff
@@ -115,14 +115,14 @@ module.exports = {
...
-    enabled: function (config) { return !!config.get('datasource_search_twitter'); },
+    enabled: function (config, userModel) { return userModel.get('twitter').enabled && !!config.get('datasource_search_twitter'); },
     fallback: require('./fallbacks/import-twitter-fallback.tpl'),
     name: 'twitter',
     title: 'Twitter'
   },
   Mailchimp: {
     view: ImportServiceView,
-    enabled: function (config) { return !!config.get('oauth_mailchimp'); },
+    enabled: function (config, userModel) { return userModel.get('mailchimp').enabled && !!config.get('oauth_mailchimp'); },
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/components/modals/add-layer/content/imports/import-options.js]: Why do you use !! here and not in the rest of checkers?

File: lib/assets/javascripts/cartodb3/components/modals/add-layer/content/imports/import-options.js [hunk 5]
```diff
@@ -24,7 +24,7 @@ module.exports = {
 
   File: {
     view: ImportDataView,
-    enabled: function (config) { return true; },
+    enabled: function (config, userModel) { return true; },
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/components/modals/add-layer/content/imports/import-options.js]: Not needed to specify any param if they are not used.

File: lib/assets/javascripts/cartodb3/components/modals/add-layer/content/imports/import-options.js [hunk 6]
```diff
@@ -35,7 +35,7 @@ module.exports = {
   },
   GDrive: {
     view: ImportServiceView,
-    enabled: function (config) { return !!config.get('oauth_gdrive'); },
+    enabled: function (config, userModel) { return !!config.get('oauth_gdrive'); },
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/components/modals/add-layer/content/imports/import-options.js]: Not needed to add/specify userModel if it is not used.

File: lib/assets/javascripts/cartodb3/components/modals/add-layer/content/imports/import-options.js [hunk 7]
```diff
@@ -58,7 +58,7 @@ module.exports = {
   },
   Dropbox: {
     view: ImportServiceView,
-    enabled: function (config) { return !!config.get('oauth_dropbox'); },
+    enabled: function (config, userModel) { return !!config.get('oauth_dropbox'); },
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/components/modals/add-layer/content/imports/import-options.js]: Same here ^^^.

File: lib/assets/javascripts/cartodb3/components/modals/add-layer/content/imports/import-options.js [hunk 8]
```diff
@@ -88,7 +88,7 @@ module.exports = {
   },
   Box: {
     view: ImportServiceView,
-    enabled: function (config) { return !!config.get('oauth_box'); },
+    enabled: function (config, userModel) { return !!config.get('oauth_box'); },
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/components/modals/add-layer/content/imports/import-options.js]: Dixit ^^^.

File: lib/assets/javascripts/cartodb3/components/modals/add-layer/content/imports/import-options.js [hunk 9]
```diff
@@ -165,14 +165,14 @@ module.exports = {
   // },
   Arcgis: {
     view: ImportArcGISView,
-    enabled: function (config) { return config.get('arcgis_enabled'); },
+    enabled: function (config, userModel) { return config.get('arcgis_enabled'); },
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/components/modals/add-layer/content/imports/import-options.js]: 💃  ^^^

File: lib/assets/javascripts/cartodb/common/dialogs/create/listing/import_options.js [hunk 10]
```diff
@@ -57,7 +57,7 @@ module.exports = {
   },
   Dropbox: {
     className: ImportService,
-    enabled: function (config) { return !!config.get('oauth_dropbox'); },
+    enabled: function (config, user_data) { return !!config.get('oauth_dropbox'); },
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb/common/dialogs/create/listing/import_options.js]: If it is not needed, don't specify it.

File: lib/assets/javascripts/cartodb/common/dialogs/create/listing/import_options.js (hunk 2 above)
Comment (xavijam) [inline lib/assets/javascripts/cartodb/common/dialogs/create/listing/import_options.js]: If it is not needed, don't specify it.

File: lib/assets/javascripts/cartodb/common/dialogs/create/listing/import_options.js [hunk 11]
```diff
@@ -23,7 +23,7 @@ var ImportArcGISView = require('./imports/import_arcgis_view');
 module.exports = {
   File: {
     className: ImportDataView,
-    enabled: function (config) { return true; },
+    enabled: function (config, user_data) { return true; },
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb/common/dialogs/create/listing/import_options.js]: If those params are not needed, don't specify them.

File: lib/assets/javascripts/cartodb/common/dialogs/create/listing/import_options.js [hunk 12]
```diff
@@ -164,14 +164,14 @@ module.exports = {
...
-    enabled: function (config) { return config.get('arcgis_enabled'); },
+    enabled: function (config, user_data) { return config.get('arcgis_enabled'); },
     fallback: 'common/views/create/listing/import_arcgis_fallback',
     name: 'arcgis',
     title: 'ArcGIS Server&trade;'
   },
   Salesforce: {
     className: ImportDataView,
-    enabled: function (config) { return config.get('salesforce_enabled'); },
+    enabled: function (config, user_data) { return config.get('salesforce_enabled'); },
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb/common/dialogs/create/listing/import_options.js]: If it is not needed, don't specify it.

File: NEWS.md (hunk 1 above)
Comment (juanignaciosl) [inline NEWS.md]: Outch, I often write files where this is needed :_)

File: lib/assets/javascripts/cartodb3/components/modals/add-layer/content/imports/import-options.js (hunk 4 above)
Comment (juanignaciosl) [inline lib/assets/javascripts/cartodb3/components/modals/add-layer/content/imports/import-options.js]: That's a good question. That's the old one, [it was like that when I got here](http://www.opinionatedbastard.ca/wp-content/uploads/2014/04/3pen1a.jpg). As new parameter is always there and boolean it's not needed, isn't it?

File: lib/assets/javascripts/cartodb3/components/modals/add-layer/content/imports/import-options.js (hunk 5 above)
Comment (juanignaciosl) [inline lib/assets/javascripts/cartodb3/components/modals/add-layer/content/imports/import-options.js]: Although it's not used, all `enabled` methods must have the same signature, as `imports-view.js` loops through them. It's needed for Mailchimp and Twitter, which use it.

File: lib/assets/javascripts/cartodb3/components/modals/add-layer/content/imports/import-options.js (hunk 6 above)
Comment (juanignaciosl) [inline lib/assets/javascripts/cartodb3/components/modals/add-layer/content/imports/import-options.js]: Same.

File: lib/assets/javascripts/cartodb3/components/modals/add-layer/content/imports/import-options.js (hunk 7 above)
Comment (juanignaciosl) [inline lib/assets/javascripts/cartodb3/components/modals/add-layer/content/imports/import-options.js]: Same.

File: lib/assets/javascripts/cartodb3/components/modals/add-layer/content/imports/import-options.js (hunk 8 above)
Comment (juanignaciosl) [inline lib/assets/javascripts/cartodb3/components/modals/add-layer/content/imports/import-options.js]: Same.

File: lib/assets/javascripts/cartodb3/components/modals/add-layer/content/imports/import-options.js (hunk 9 above)
Comment (juanignaciosl) [inline lib/assets/javascripts/cartodb3/components/modals/add-layer/content/imports/import-options.js]: Same

File: lib/assets/javascripts/cartodb3/components/modals/add-layer/content/imports/import-options.js [hunk 13]
```diff
@@ -165,14 +165,14 @@ module.exports = {
...
     fallback: require('./fallbacks/import-arcgis-fallback.tpl'),
     name: 'arcgis',
     title: 'ArcGIS Server&trade;'
   },
   Salesforce: {
     view: ImportDataView,
-    enabled: function (config) { return config.get('salesforce_enabled'); },
+    enabled: function (config, userModel) { return config.get('salesforce_enabled'); },
```
Comment (juanignaciosl) [inline lib/assets/javascripts/cartodb3/components/modals/add-layer/content/imports/import-options.js]: Same. Next time once is enough :P

File: lib/assets/javascripts/cartodb/common/dialogs/create/listing/import_options.js (hunk 10 above)
Comment (juanignaciosl) [inline lib/assets/javascripts/cartodb/common/dialogs/create/listing/import_options.js]: Same

File: lib/assets/javascripts/cartodb/common/dialogs/create/listing/import_options.js [hunk 14]
```diff
@@ -87,7 +87,7 @@ module.exports = {
   },
   Box: {
     className: ImportService,
-    enabled: function (config) { return !!config.get('oauth_box'); },
+    enabled: function (config, user_data) { return !!config.get('oauth_box'); },
```
Comment (juanignaciosl) [inline lib/assets/javascripts/cartodb/common/dialogs/create/listing/import_options.js]: Same

File: lib/assets/javascripts/cartodb/common/dialogs/create/listing/import_options.js (hunk 11 above)
Comment (juanignaciosl) [inline lib/assets/javascripts/cartodb/common/dialogs/create/listing/import_options.js]: Same >_<

File: lib/assets/javascripts/cartodb/common/dialogs/create/listing/import_options.js [hunk 15]
```diff
@@ -164,14 +164,14 @@ module.exports = {
   // },
   Arcgis: {
     className: ImportArcGISView,
-    enabled: function (config) { return config.get('arcgis_enabled'); },
+    enabled: function (config, user_data) { return config.get('arcgis_enabled'); },
```
Comment (juanignaciosl) [inline lib/assets/javascripts/cartodb/common/dialogs/create/listing/import_options.js]: Same >_<

File: lib/assets/javascripts/cartodb/common/dialogs/create/listing/import_options.js (hunk 12 above)
Comment (juanignaciosl) [inline lib/assets/javascripts/cartodb/common/dialogs/create/listing/import_options.js]: Same >_<

**Comment by juanignaciosl** [comment]
@xavijam next CR, please. If an issue is repeated, saying it once or twice should be enough :P

**Comment by juanignaciosl** [comment]
@gfiorav in a last moment double check I realized that I was using `@user` within `User` decorator, which is wrong. 👀  before merging, please?

**Pull request merged**